src/psyclone/tests/nemo/transformations/openmp/openmp_test.py

```diff
@@ -147,9 +147,80 @@ def test_omp_parallel_multi():
     # stored AST.
     directive = new_sched.children[0].loop_body[2]
     assert isinstance(directive, OMPParallelDirective)
-    old_ast = directive._ast
+    old_ast = directive.ast
     directive.update()
-    assert old_ast is directive._ast
+    assert old_ast is directive.ast
+
+
+def test_omp_do_update():
+    '''Check the OMPDoDirective update function.'''
+    from psyclone.transformations import OMPLoopTrans, OMPParallelTrans
+    from psyclone.psyGen import OMPDoDirective
+    _, invoke_info = parse(os.path.join(BASE_PATH, "imperfect_nest.f90"),
+                           api=API, line_length=False)
+    psy = PSyFactory(API, distributed_memory=False).create(invoke_info)
+    schedule = psy.invokes.get('imperfect_nest').schedule
+    par_trans = OMPParallelTrans()
+    loop_trans = OMPLoopTrans()
+    new_sched, _ = par_trans.apply(schedule.children[0].loop_body[1]
+                                   .else_body[0].else_body[0])
+    new_sched, _ = loop_trans.apply(new_sched.children[0].loop_body[1]
+                                    .else_body[0].else_body[0].children[0])
+    gen_code = str(psy.gen).lower()
+    correct = '''      !$omp parallel default(shared), private(ji,jj)
+      !$omp do schedule(static)
+      do jj = 1, jpj, 1
+        do ji = 1, jpi, 1
+          zdkt(ji, jj) = (ptb(ji, jj, jk - 1, jn) - ptb(ji, jj, jk, jn)) * \
+wmask(ji, jj, jk)
+        end do
+      end do
+      !$omp end do
+      !$omp end parallel'''
+    assert correct in gen_code
+
+    # Call update a second time and make sure that this does not
+    # trigger the whole update process again.
+    directive = new_sched.children[0].loop_body[1].else_body[0].else_body[0]\
+        .children[0]
+    assert isinstance(directive, OMPDoDirective)
+    directive.update()
```

Comment:
Shouldn't this second call to `update()` be done before the check on the structure of the PSyIR? As it stands, the test immediately wipes the existing AST after this call to `update().`

Reply:
Yes. that code was somehow messed up ... and didn't test anything. I added a test that the old and new AST are actually the same.